client/dashboard/app/hooks/site-logs-data.ts

```diff
@@ -0,0 +1,76 @@
+import { useQuery } from '@tanstack/react-query';
+import { LogType, PHPLog, ServerLog, FilterType } from '../../data/site-logs';
+import { siteLogsQuery } from '../queries/site-logs';
+import type { View } from '@wordpress/dataviews';
+
+const EMPTY_ARRAY: ( ServerLog | PHPLog )[] = [];
+
+const hasLogTypeChanged = ( logType: LogType, data: { logs?: ( PHPLog | ServerLog )[] } ) => {
+	if (
+		logType === 'php' &&
+		!! data.logs &&
+		data.logs.length > 0 &&
+		'severity' in data.logs[ 0 ] &&
+		'message' in data.logs[ 0 ]
+	) {
+		return false;
+	}
+	if (
+		logType === 'server' &&
+		!! data.logs &&
+		data.logs.length > 0 &&
+		'request_type' in data.logs[ 0 ] &&
+		'status' in data.logs[ 0 ]
+	) {
+		return false;
+	}
+	return true;
+};
+
+export function useSiteLogsData( {
```

Comment:
Can we put inline this logic in the component?

Basically we want to avoid sharing hooks like this, unless absolutely necessary.

Reply:
I can do, yes. Done in f0d878e00ca1004f402376cbb0b0e4e031fe2919. I have to wrap my head around not defaulting to separating hooks.